src/freenet/client/async/SplitFileInserter.java

```diff
@@ -175,6 +175,7 @@ public void onFinishedEncode() {
 
     @Override
     public void encodingProgress() {
+        if(ctx.getCHKOnly) return;
```

Comment:
We have already generated the blocks. start() would check that we've generated the blocks and then attempt to actually insert. The check does disk I/O, so this is unnecessary cost, and IIRC we've already closed the file at this point in this case.